services/frontend-v3/src/pages/Profile.jsx

```diff
@@ -9,66 +9,124 @@ const Profile = ({ history, match }) => {
-        .catch((error) => {
-          if (
-            !error.isAxiosError ||
-            !error.response ||
-            !error.response.status === 404
-          ) {
-            throw error;
-          }
-        });
-      return fetchedData;
-    },
-    [userID, locale]
-  );
+      promiseArray.push(profile);
 
-  useEffect(() => {
-    const { id } = match.params;
+      if (id !== userID) {
+        const connectionStatus = axios.get(`api/connections/${id}`);
+        promiseArray.push(connectionStatus);
+      }
 
-    if (id === undefined) {
-      history.push(`/secured/profile/${userID}`);
-    } else {
-      updateProfileInfo(id)
-        .then((fetchedData) => {
-          if (fetchedData !== undefined) {
-            setName(`${fetchedData.firstName} ${fetchedData.lastName}`);
+      Promise.all(promiseArray)
+        .then((result) => {
+          if (result[0].data !== undefined) {
+            const profileData = result[0].data;
+            setName(`${profileData.firstName} ${profileData.lastName}`);
+            setData(profileData);
+            if (result[0].data && userID !== id) {
+              const isConnection = result[0].data.connections
+                ? result[0].data.connections.some((item) => item.id === userID)
+                : false;
+              const connectionStatus = result[1].data.status;
+              const {
+                info,
+                manager,
+                projects,
+                skills,
+                competencies,
+                education,
+                experience,
+                exFeeder,
+              } = profileData.visibleCards;
+              profileData.visibleCards = {
+                info: !(
+                  info === "PRIVATE" ||
+                  (info === "CONNECTIONS" && !isConnection)
+                ),
+                manager: !(
+                  manager === "PRIVATE" ||
+                  (manager === "CONNECTIONS" && !isConnection)
+                ),
+                projects: !(
+                  projects === "PRIVATE" ||
+                  (projects === "CONNECTIONS" && !isConnection)
+                ),
+                skills: !(
+                  skills === "PRIVATE" ||
+                  (skills === "CONNECTIONS" && !isConnection)
+                ),
+                competencies: !(
+                  competencies === "PRIVATE" ||
+                  (competencies === "CONNECTIONS" && !isConnection)
+                ),
+                education: !(
+                  education === "PRIVATE" ||
+                  (education === "CONNECTIONS" && !isConnection)
+                ),
+                experience: !(
+                  experience === "PRIVATE" ||
+                  (experience === "CONNECTIONS" && !isConnection)
+                ),
+                exFeeder: !(
+                  exFeeder === "PRIVATE" ||
+                  (exFeeder === "CONNECTIONS" && !isConnection)
+                ),
+              };
```

Comment:
Potentially do this on the server? I saw something similar in our APIs, aren't we already doing this? If you want to keep it on the client, I would suggest the same thing as above, to extract the logic to a function